### PR TITLE
[docs] Address charts docs feedback

### PR DESCRIPTION
**Pull request by alexfauquette**
Fix two docs feedback

> direction property in the legend object is misspelled.
sent from https://mui.com/x/react-charts/legend/ (from section [Placement)](https://mui.com/x/react-charts/legend/#placement)

> It's not obvious that this is a prop for 'series', I thought it was a base prop for LineChart.
I was only able to see the correct placement after inspecting the element on the page to see how the props were written.
sent from https://mui.com/x/react-charts/lines/ (from section [Interpolation)](https://mui.com/x/react-charts/lines/#interpolation)

<table>
<tr>
 <td><a href="https://mui.com/x/react-charts/lines/#interpolation">Before</a></td>
 <td><a href="https://deploy-preview-10119--material-ui-x.netlify.app/x/react-charts/lines/#interpolation">After</a></td>
 </tr>
<tr>
 <td>
 <td>
 </tr>
</table>

### CODE REVIEW

File: docs/data/charts/lines/InterpolationDemoNoSnap.js [hunk 1]
```diff
@@ -15,7 +17,16 @@ const curveTypes = [
   'stepAfter',
 ];
 
-export default function InterpolationDemo() {
+function getExample(curveType) {
+  return `<LineChart
+  series={[
+    { curve: "${curveType}", data: [0, 5, 2, 6, 3, 9.3] },
```
Comment (LukasTy) [inline docs/data/charts/lines/InterpolationDemoNoSnap.js:23]: Have you considered setting only one of the series `curve` to the controlled one to showcase that different series can have different curve types? 🤔

**Review by LukasTy — APPROVED** [review]
Nice improvement! 💯

**Comment by LukasTy** [comment]
BTW, should we change the PR title to `[docs] Address charts docs feedbacks` so that this change doesn't end up as a change of the package itself? 🤔

File: docs/data/charts/lines/lines.md [hunk 2]
```diff
@@ -59,7 +59,9 @@ For more information, see [stacking docs](/x/react-charts/stacking/).
 The interpolation between data points can be customized by the `curve` property.
 This property expects one of the following string values, corresponding to the interpolation method: `'catmullRom'`, `'linear'`, `'monotoneX'`, `'monotoneY'`, `'natural'`, `'step'`, `'stepBefore'`, `'stepAfter'`.
 
-{{"demo": "InterpolationDemo.js", "hideToolbar": true, "bg": "inline"}}
+This is a series' property such that different series can have different interpolations.
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply docs/data/charts/lines/lines.md:62]:
```suggestion
This series property adds the option to control the interpolation of a series.
Different series could even have different interpolations.
```

File: docs/data/charts/lines/InterpolationDemoNoSnap.js (hunk 1 above)
Comment (LukasTy) [reply docs/data/charts/lines/InterpolationDemoNoSnap.js:23]: Gotcha. I thought that having this implemented in the demo would complement the updated paragraph text very well.
But if you think that it is not worth the effort/confusion, feel free to leave as is. 👌

**Pull request merged**